pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -288,7 +287,14 @@ private SuccessResponse uploadSegment(FormDataMultiPart multiPart, boolean enabl
           throw new UnsupportedOperationException("Unsupported upload type: " + uploadType);
       }
 
-      String rawTableName = segmentMetadata.getTableName();
+      // Fetch raw table name. Try to derive the table name from the parameter and then from segment metadata
+      String rawTableName = null;
+      if (tableName != null && !tableName.isEmpty()) {
+        rawTableName = TableNameBuilder.extractRawTableName(tableName);
```

Comment:
It'd be good to add some log messages as this API won't be called frequently.

Reply:
added logs.